custom_components/waste_collection_schedule/waste_collection_schedule/source/lipizzanerheimat_at.py

```
@@ -0,0 +1,333 @@
+    def sha1(input_str):
+        return hashlib.sha1(input_str.encode('utf-8')).hexdigest()
+
+    def api_hash(self, endpoint, payload):
+        return self.sha1(f"{APISECRET}@{endpoint}@{payload}")
+
+    @staticmethod
+    def get_basic_auth_header(user, password):
+        credentials = f"{user}:{password}"
+        base64_credentials = base64.b64encode(
+            credentials.encode("utf-8")).decode("utf-8")
+        return f"Basic {base64_credentials}"
+
+    def sha1(self, input_str):
+        return hashlib.sha1(input_str.encode('utf-8')).hexdigest()
+
+    def build_header(self, url, payload):
+        apihash = self.api_hash(url, payload)
+        headers = self.default_headers
+        headers['apihash'] = apihash
+        return headers
+
+    def fetch_communities(self):
+
+        url = 'api/mobile-app/getcommunities'
+        payload = 'language=de'
+
+        headers = self.build_header(url, payload)
+
+        communities = requests.request(
+            "POST", f'{API_URL}/{url}', headers=headers, data=payload).json()['data']
+
+        return next((c['community_id'] for c in communities if c['community_title'] == self.town), None)
+
+    def fetch_calendar_id_by_name(self):
+
+        url = 'api/mobile-app/garbagecalendarindexwithstreets'
+        payload = f'language=de&communities=234|110|110|111|112|241|238|119|113|236|114|237|115|116|117|235&garbage_calendar_id=1'
+
+        headers = self.build_header(url, payload)
+
+        calendarData = requests.request(
+            "POST", f'{API_URL}/{url}', headers=headers, data=payload).json()['data']
+
+        return next((c['garbage_calendar_id']
+                     for c in calendarData['garbage_calendars'] if c['garbage_calendar_title'] == self.map_name), None)
+
+    def fetch_calendar_id(self):
+
+        community_id = self.fetch_communities()
+
+        if (community_id == None):
+            raise SourceArgumentNotFound(
+                [], "Town/Street combination not found.")
```

This exception does not work like this. (see above)

You could raise the SourceArgumentNotFoundWithSuggestions exception in the fetch_communities function to provide feedback which towns would be valid